lib/blacklight.rb

```diff
@@ -25,6 +23,20 @@ class << self
     attr_accessor :solr, :solr_config
   end
 
+  # For deprecating SolrRepository and SolrResponse
+  def self.const_missing(const_name)
+    case const_name
```

Comment:
Instead of using `const_missing`, could we define and deprecate them like we did for modules?

https://github.com/projectblacklight/blacklight/commit/3ed993a53d8f9c2815284bc354f6c8a113cb5630